assets/js/googlesitekit/data/create-error-store.test.js

```diff
@@ -0,0 +1,221 @@
+			} );
+		} );
+
+		describe( 'clearError', () => {
+			it( 'removes an error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				dispatch.clearError( ERROR_SAMPLE.baseName );
+
+				const errorObj = {};
+				errorObj[ ERROR_SAMPLE_2.baseName ] = ERROR_SAMPLE_2.error;
+				expect( store.getState().errors ).toMatchObject( errorObj );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getErrorForSelector', () => {
+			it( 'requires a selector param', () => {
+				expect( () => {
+					select.getErrorForSelector();
+				} ).toThrow( 'selectorName is required.' );
+			} );
+
+			it( 'returns undefined when error does not exist', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const selectedError = select.getErrorForSelector( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( selectedError ).toBeUndefined();
+			} );
+
+			it( 'returns the required error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const selectedError = select.getErrorForSelector( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getErrorForAction', () => {
+			it( 'requires a selector param', () => {
+				expect( () => {
+					select.getErrorForAction();
+				} ).toThrow( 'actionName is required.' );
+			} );
+
+			it( 'returns undefined when error does not exist', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( selectedError ).toBeUndefined();
+			} );
+
+			it( 'returns the specified error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getError', () => {
+			it( 'requires a baseName param', () => {
+				select.getError();
+
+				expect( store.getState().error ).toBeUndefined();
+			} );
+
+			it( 'returns the correct error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				const error = select.getError( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( error ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
```

Comment:
We need to cover the following here:
* `state.error` is returned when `getError()` is called with no arguments
* The appropriate error is returned when called _with_ a `baseName` and `args`